tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs

```diff
@@ -199,21 +278,22 @@ public void ABeaconSendingCaptureOnStateContinuesWithNextFinishedSessionIfSendin
             //given
             var target = new BeaconSendingCaptureOnState();
 
-            var finishedEmptySession = CreateEmptySession("127.0.0.2");
-            var finishedSession = CreateValidSession("127.0.0.1");
+            var sessionOne = new SessionWrapper(CreateEmptySession("127.0.0.2"));
+            var sessionTwo = new SessionWrapper(CreateValidSession("127.0.0.1"));
+            finishedSessions.AddRange(new[] { sessionOne, sessionTwo });
 
             var statusResponses = new Queue<StatusResponse>();
-            context.GetNextFinishedSession().Returns(finishedEmptySession, finishedSession, null);
             httpClient.SendBeaconRequest(Arg.Any<string>(), Arg.Any<byte[]>()).Returns(new StatusResponse(string.Empty, 200));
-            
+
             //when calling execute
             target.Execute(context);
 
-            context.Received(3).GetNextFinishedSession();
-            context.Received(0).PushBackFinishedSession(finishedSession);
+            var tmp = context.Received(1).FinishedAndConfiguredSessions;

```

Comment:
`tmp` is never used

Reply:
Unfortunately an NSubstitute limitation http://nsubstitute.github.io/help/received-calls/